### ISSUE TITLE
Broken contact request notification appears for contact request sender if contact request receiver starts sending messages before sender receives signal about contact request acception

### ISSUE THREAD

**Issue by pavloburykh** [status-im/status-react#21438]
This is a very old issue. Haven't found it in repo so logging it now. In case you will find a duplicate feel free to close this one

**Steps:** 

1. User A sends contact request to User B
2. Go offline by User A. (NOTE: we need to do this in order to easily generate the required condition. In real life it can happen due to network problems or issues on Waku side. 
4. Accept contact request by User B. 
5. Send a stack of messages in 1-1 chat by User B
6. Come back online by User A
7. Check activity center for notifications by User A when connection reestablished 

**Actual result:** notification with broken contact request from User B appears in Activity center for User A. In fact it is not even contact request as User B has not sent any requests but accepted it from User B. What happens here is in case of delay of receiving by User A the signal about User B has accepted his CR, the messages sent by User B turn to appear as such broken contact requests notifications in activity center for User A (as User B is not a contact for User A yet despite he has already accepted request and is able to send messages). Such notification cannot be deleted as it is not responsive for swipes. It can be read only by using Mark all as read function. 

[User__B_cr_receiver.zip](https://github.com/user-attachments/files/17381294/User__B_cr_receiver.zip)
[User_A_cr_sender.zip](https://github.com/user-attachments/files/17381293/User_A_cr_sender.zip)
![Image](https://github.com/user-attachments/assets/0c465f85-9064-4fdc-8354-9045883046a3)

https://github.com/user-attachments/assets/91a25413-165c-4ba0-9f7b-7663612524d0

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Comment by churik** [comment]
From discussion with @ilmotta 
It shouldn't be possible from the user perspective to do 4 step without acknowledgement from the receiver side.

That's something for both clients to look into.

**Comment by churik** [comment]
I don't think that it is feasible to finish investigation in 2.32, given the desired release cut date.
Deferring the issue, please be free to move it back if you think there is a room for it @ilmotta

**Comment by ilmotta** [comment]
@pavloburykh what's your take from this comment https://github.com/status-im/status-mobile/issues/21438#issuecomment-2419422325 ?

**Comment by pavloburykh** [comment]
> [@pavloburykh](https://github.com/pavloburykh) what's your take from this comment [#21438 (comment)](https://github.com/status-im/status-mobile/issues/21438#issuecomment-2419422325) ?

@ilmotta @churik from one point, solution proposed by Tetiana would fix current problem. From user perspective the solution is not ideal, as it would be confusing for me, as contact request receiver why I cannot send messages right after accepting incoming contact request. I believe input from design team would be helpful here as well. cc @xAlisher

**Comment by ilmotta** [comment]
> > [@pavloburykh](https://github.com/pavloburykh) what's your take from this comment [#21438 (comment)](https://github.com/status-im/status-mobile/issues/21438#issuecomment-2419422325) ?
> 
> [@ilmotta](https://github.com/ilmotta) [@churik](https://github.com/churik) from one point, solution proposed by Tetiana would fix current problem. From user perspective the solution is not ideal, as it would be confusing for me, as contact request receiver why I cannot send messages right after accepting incoming contact request. I believe input from design team would be helpful here as well. cc [@xAlisher](https://github.com/xAlisher)

I see now what you mean @pavloburykh about the UX part. Most users would expect being able to immediately send messages after accepting a CR, as this is natural in centralized messengers where a central server is always online on behalf of users.

Thinking deeper now, I believe the best experience is if Alice and Bob can exchange messages without ever caring about each other's online state. Anything other than that and we might create a fragile solution because Alice and Bob can both be in highly intermittent connections. Sometimes the best is to solve for the worst case, which will cover all cases.

@jrainville what's your take on this problem? There's probably a way to simulate the case of this issue in a test in status-go and derive a more robust solution.

**Comment by jrainville** [comment]
@ilmotta @pavloburykh @churik I agree that waiting for an acknowledgement from User 1 is not a good idea. It would add a lot of friction and be bad UX.

Quick question: the messages themselves all appear in the 1-1 chat right? I just want to be sure we don't have a bigger issue on our hands. Though I'm pretty sure this is something that would get fixed automatically by MVDS (waku side resend)

As for the AC issue, I might have just pushed a fix on status-go that helps or fixes that issue. Funny timing once again 😄 
https://github.com/status-im/status-go/pull/6140

My PR is not about this exact reproduction steps, but it's related. What happens is that User 1 sends a CR and User 2 dismisses it (nothing is sent, this is local). Afterwards, User 2 sends back a new CR. This new CR is the same as the message "1" sent in the video.
So before, the new CR sent by User 2 would appear in the AC, but I made it appear in the chat instead and I mark the old AC notification from User 1 as accepted (sending CR from both side is the same as CR + Accept).

Let me know if I was clear 😅 

@pavloburykh if you could test that PR, we could confirm if it fixes your issue as well 😃

**Comment by pavloburykh** [comment]
> [@pavloburykh](https://github.com/pavloburykh) if you could test that PR, we could confirm if it fixes your issue as well 😃

@jrainville thank you. Sure, I can check on Monday. Could you please rebase [status go PR](https://github.com/status-im/status-go/pull/6140) as it is quite outdated? After that we will create corresponding mobile PR to test changes.

**Comment by pavloburykh** [comment]
@jrainville thank you. Looks like https://github.com/status-im/status-go/pull/6140 fixes current issue. Tested in https://github.com/status-im/status-mobile/pull/21729.

**Comment by pavloburykh** [comment]
closing as fixed by https://github.com/status-im/status-go/pull/6140
